### PR TITLE
Refactor CollectionScreen to Implement MVVM Architecture with ViewModel and Repository

### PR DESCRIPTION
**Pull request by othbcq**
### PR Description
This PR refactors `CollectionScreen` to align with MVVM architecture, improving modularity, testability, and maintainability by separating data handling from UI logic.

#### Key Changes:

1. **Repository**: Added `CollectionRepository` for data access and implemented `CollectionRepositoryFirestore` to manage Firebase Storage interactions, with improved error handling and logging.

2. **ViewModel**: Introduced `CollectionViewModel` to handle image data, exposing `imageUrls` for the UI. Manages loading, success, and error states, enhancing separation of concerns.

3. **Composable Update**: Refactored `CollectionScreen` to observe `imageUrls` from the `ViewModel` instead of directly accessing Firebase, and added UI elements for loading and error states.

4. **Testing**: Implemented tests for `CollectionRepositoryFirestore` and `CollectionViewModel`, covering edge cases and state handling. Added UI tests for `CollectionScreen` to verify image loading and interactions.

### CODE REVIEW

File: app/src/main/java/com/github/lookupgroup27/lookup/model/collection/CollectionViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  init {
+    // Automatically fetch images when the ViewModel is created
+    viewModelScope.launch { fetchImages() }
+  }
+
+  private suspend fun fetchImages() {
+    val images = repository.getUserImageUrls()
+    _imageUrls.value = images
+  }
```
Comment (Ismaillat) [inline app/src/main/java/com/github/lookupgroup27/lookup/model/collection/CollectionViewModel.kt:25]: You may need to handle potential errors in this function.

**Review by Ismaillat — CHANGES_REQUESTED** [review]
Everything looks good to me, except in the fetchImages function in which you do not handle potential errors. 
You have taken into  account the edge cases and have a good line coverage.
Good Job!

**Review by Ismaillat — APPROVED** [review]
Looks good to me. Well done !
